### PR TITLE
remove component enums in favor of const objects

### LINKED ISSUE #5765: rfc: Deprecate Custom Element Attribute Enums?
https://github.com/microsoft/fast/pull/5745 introduces string literal type unions for custom element HTML attributes where they exist in the FAST component packages. 

This issue tracks an open question from that PR is if the enum exports that exist for these attributes should be deprecated.

### PR DESCRIPTION
**Pull request by chrisdholt**
# Pull Request

## 📖 Description
This PR updates existing enums for components and converts them to a const object + supported type values.

### 🎫 Issues
fixes #5765

### General

- [x] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

File: packages/web-components/fast-foundation/src/accordion/README.md [hunk 1]
```diff
@@ -147,6 +147,20 @@ export const myAccordionItem = AccordionItem.compose<AccordionItemOptions>({
...
+
+| Name                  | Description               | Type                                                                                                                                                                                                                                                                                                                             |
+| --------------------- | ------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `AccordionExpandMode` | Expand mode for Accordion | `{
+    /**      * Designates only a single {@link @microsoft/fast-foundation#(AccordionItem:class) } can be open a time.      */
+    single: "single",
+
+    /**      * Designates multiple {@link @microsoft/fast-foundation#(AccordionItem:class) or AccordionItems} can be open simultaneously.      */
+    multi: "multi",
+}` |
```
Comment (radium-v) [inline packages/web-components/fast-foundation/src/accordion/README.md:160]: The formatting for these will need to be handled in the `CEMToMarkdown` script

Comment (chrisdholt) [reply packages/web-components/fast-foundation/src/accordion/README.md:160]: I think it should be but definitely odd that it's not. I'll try running everything again locally - not sure why this wouldn't transform.

Comment (chrisdholt) [reply packages/web-components/fast-foundation/src/accordion/README.md:160]: OH - These are the new values coming in. I'll take a look before tagging Bill :)

Comment (chrisdholt) [reply packages/web-components/fast-foundation/src/accordion/README.md:160]: @williamw2 Looks like we may need some custom handling to enable proper docs for moving from enums to const objects - is that something you might be able to help with?

**Review by nicholasrice — APPROVED** [review]

File: packages/web-components/fast-foundation/src/accordion/README.md (hunk 1 above)
Comment (williamw2) [reply packages/web-components/fast-foundation/src/accordion/README.md:160]: I'll take a look.

Comment (chrisdholt) [reply packages/web-components/fast-foundation/src/accordion/README.md]: Per @williamw2 I think we'll need some kind of additional formatting solution here. I don't think we should block on this, but we should figure out that solution to resolve the formatting issue.

**Review by EisenbergEffect — APPROVED** [review]

**Comment by KingOfTac** [comment]
Just out of curiosity, I'm wondering why const objects are more preferable to enums.

**Comment by chrisdholt** [comment]
> Just out of curiosity, I'm wondering why const objects are more preferable to enums.

There are a few reasons, but a big one is that there isn't (yet) the concept of enums in JavaScript and we've had feedback from folks who have ran into difficulty not being able to leverage certain values because of that. Additionally, enums don't really compress, so while a few may not cause much distress, creating a pattern around enums can result in "death by 1000 cuts" on the performance side of things. There is a pretty exhaustive resource here from Typescript on Enums which is worth a look; deep linking specifically to the objects vs. enums example but the whole doc is worth reading IMO: https://www.typescriptlang.org/docs/handbook/enums.html#objects-vs-enums

**Pull request merged**